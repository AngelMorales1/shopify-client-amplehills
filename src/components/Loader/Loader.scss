@import 'settings';

.Loader {
  justify-content: start;

  @media ($medium-up) {
    justify-content: center;
  }

  &__bar {
    height: spacing(1);
    width: 100%;
    background: color('peach');
    animation: move 10s linear;
    left: -15%;

    @media ($medium-up) {
      height: spacing(2);
    }
  }

  &__image {
    @media ($medium-up) {
      width: 110px;
    }
  }
}

@keyframes move {
  0% {
    left: -100%;
  }
  50% {
    left: -40%;
  }
  70% {
    left: -20%;
  }
  75% {
    left: -16%;
  }
  90% {
    left: -15%;
  }
  100% {
    left: -15%;
  }
}
