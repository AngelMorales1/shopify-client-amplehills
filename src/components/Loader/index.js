import React from 'react';
import cx from 'classnames';

import { Image } from 'components/base';
import styles from './Loader.scss';

const Loader = () => {
  return (
    <div className={cx(styles['Loader'], `w100 flex bg-white p2`)}>
      <Image
        className={styles['Loader__image']}
        src="/assets/images/ample-hills-logo.svg"
      />
    </div>
  );
};

export default Loader;
