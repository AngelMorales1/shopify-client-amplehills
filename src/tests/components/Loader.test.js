import React from 'react';
import { shallow } from 'enzyme';

import Loader from 'components/Loader';

it('renders without data', () => {
  const component = shallow(<Loader />);

  expect(component).toMatchSnapshot();
});
