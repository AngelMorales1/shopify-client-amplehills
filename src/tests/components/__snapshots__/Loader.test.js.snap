// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Image
        alt=""
        bg={false}
        className="w100"
        onImgLoad={[Function]}
        src="/assets/images/ample-hills-logo.svg"
        style={Object {}}
/>,
      "className": "w100 flex bg-white p2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alt": "",
        "bg": false,
        "children": null,
        "className": "w100",
        "onImgLoad": [Function],
        "src": "/assets/images/ample-hills-logo.svg",
        "style": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Image
          alt=""
          bg={false}
          className="w100"
          onImgLoad={[Function]}
          src="/assets/images/ample-hills-logo.svg"
          style={Object {}}
/>,
        "className": "w100 flex bg-white p2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "",
          "bg": false,
          "children": null,
          "className": "w100",
          "onImgLoad": [Function],
          "src": "/assets/images/ample-hills-logo.svg",
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
